scripts/bundle/bundle-linux.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+set -euo pipefail
+# Installs the Porter CLI for a single user.
+# Environment Variables:
+# PORTER_HOME:      Location where Porter is installed (defaults to ~/.porter).
+# PORTER_MIRROR:    Base URL for downloading Porter assets (defaults to https://cdn.porter.sh).
+# PORTER_VERSION:   The version of Porter assets to download (defaults to latest).
+export PORTER_HOME=${PORTER_HOME:-~/.porter}
+export PORTER_MIRROR=${PORTER_MIRROR:-https://cdn.porter.sh}
+PORTER_VERSION=${PORTER_VERSION:-latest}
+
+MONGO_IMAGE_FILE="mongo_image.tar"
+MONGO_IMAGE_URL="docker.io/library/mongo:4.0-xenial"
+BUNDLE_NAME="porter-air-gapped-install-$PORTER_VERSION"
+DOWNLOAD_DIR="/tmp/$BUNDLE_NAME"
+
+mkdir -p $DOWNLOAD_DIR
+
+echo "Pulling MongoDB image..."
+docker pull "$MONGO_IMAGE_URL"
+echo "Saving MongoDB image to $MONGO_IMAGE_FILE..."
+docker save -o "$DOWNLOAD_DIR/$MONGO_IMAGE_FILE" "$MONGO_IMAGE_URL"
+echo "Downloading Porter from $PORTER_MIRROR/$PORTER_VERSION/install-linux.sh"
+curl -fsSLo "$DOWNLOAD_DIR/install-linux.sh" "$PORTER_MIRROR/$PORTER_VERSION/install-linux.sh"
+echo "Downloading Porter from $PORTER_MIRROR/$PORTER_VERSION/porter-linux-amd64"
```

Comment:
Why is Porter being downloaded two times? Once with `install-linux.sh` and once more directly from Github releases? It also affects the logic in the `install-linux.sh` script

Reply:
Porter is only downloaded once. The install-linux.sh has been modified to only download the `porter-linux-amd64` binary if it doesn't exist locally in the current directory. Here is a break down of how `bundle-linux.sh` would be used in practice.

1. Run `curl -L https://raw.githubusercontent.com/getporter/porter/main/scripts/bundle/bundle-linux.sh | bash` in a connected environment. This will download and bundle the assets required to install porter in an offline environment.
2. Bring the `porter-air-gapped-install-1.1.0.tar.gz` file to offline environment and untar it.
3. Run `bash install-linux.sh` and it will use the locally available files to install porter and mongo. The `install-linux.sh` script will not curl anything during the install.

This process is outlined in [docs/content/docs/getting-started/install-porter.md](https://github.com/getporter/porter/pull/3205/files#diff-3c3b63308e68595f8ea0578a6616c3045495d24e39e31bd776c600461f7ad915R114-R128)

Reply:
Ahh yes, makes sense